File: Domain/Appointment.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain
{
    //مواعيد
    public class Appointment
    {
        public Guid Id { get; set; }
        public DateTime TheAppointment { get; set; }
        public bool IsDeleted { get; set; } = false;
        public bool IsBooked { get; set; } = false;
        public Guid ServiceId { get; set; }
        public Service Service { get; set; }
        public Guid? ReservationId { get; set; }
        public Reservation? Reservation { get; set; }
    }
}

File: ScheduledServicesCenterUI/src/app/appointment/appointment.ts

import { Guid } from 'guid-typescript'; 
import { Reservation } from '../reservation/reservation';
import { Service } from '../service/service';

export interface Appointment {

    Id: Guid;
    TheAppointment: any;
    IsDeleted: boolean
    IsBooked: boolean;
    ServiceId: Guid;
    Service: Service;
    ReservationId: Guid | null;
    Reservation: Reservation| null;
}

